### DISCUSSION TITLE
Side bar menu should offer drag/drop of menu items and icon preceding menu item

### DISCUSSION THREAD

**Discussion by markpawson6** [GovAlta/ui-components#1341] [Feature request or enhancement]
- I am creating a side bar menu and expected that I could just enter my menu items in any order to get the names out of my head and then position vertically later once I know the Information Architecture. I cannot do this. I need User and Roles at the top and thought I could just drag each menu item under the layer panel to its correct place. But it appears I have to rename every menu item in the correct order.
- We use icons that precede the menu label in our app. Can we have this in a future enhancement. 

Example of the side bar menu below that I just created
![Side menu](https://github.com/GovAlta/ui-components/assets/113639558/4092ef9a-4f27-4148-9e47-a2b45cd1c4a3)

**Comment by twjeffery** [comment]
Hey Mark, do you mean in Figma? If so, you can detach the parent element and then you will be able to move the items around (and add new ones). The individual menu items are also components, so when you detach the parent those will stay intact. 

<img width="793" alt="image" src="https://github.com/GovAlta/ui-components/assets/82968683/9744a8c0-9e6d-48f7-89f7-56825c5301e3">

Let me know if you have any questions!

**Reply by twjeffery** [reply]
For the icon specifically, it would be great to chat to go through what that looks like in your service! I think it would be a relatively simple enhancement.

**Reply by markpawson6** [reply]
Here is an example from our service. In this case I will not need icons in my mock because the devs know what they are. But if the ticket I am working on had another concept that I solve using new menu items then I would need to show icons. If it worked like some of our other components that offer the ability to show text with a  icon - like a button - that would be handy.
![image](https://github.com/GovAlta/ui-components/assets/113639558/a9cef182-c546-49f4-9a04-da4dd9547607)
